R/exposure_metrics.R

```diff
@@ -132,3 +132,102 @@ calc_auc_from_regimen <- function(
     target_design$type
   )
 }
+
+#' Calculate time to target attainment
+#'
+#' For AUC targets, this function returns the first time of the start of
+#' the dosing interval plus the infusion length where the AUC24 or 12 is
+#' on target.
+#'
+#' For trough targets, this returns the time of start of the dosing interval
+#' plus the infusion length immediately prior to the trough being on target
+#' (i.e. the dose + infusion length of the first dose that brought the trough
+#' to target).
+#'
+#' @param regimen PKPDsim regimen object
+#' @param target_design target design, created using `create_target_design()`
+#' @param auc_comp compartment to look for AUC
+#' @param model PKPDsim model object
+#' @param ... arguments passed on to `PKPDsim::sim`
+#' @return a numeric value indicating the time, in hours, if a dosing interval
+#'   contained the target metric within range. Otherwise, returns NA.
```

Comment:
There are some code comments and function documentation that needs to be updated to reflect the fact that we now return a number, NA or Inf.